src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java

```diff
@@ -0,0 +1,99 @@
+package com.endava.soa_5.actions;
+
+import com.endava.soa_5.constant.LogInValidationMessages;
+import com.endava.soa_5.page_objects.ChangePasswordObjects;
+import com.endava.soa_5.page_objects.LogInPageObject;
+import com.endava.soa_5.tools.PropertyReader;
+import com.mifmif.common.regex.Generex;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.endava.soa_5.tools.Constants.PROPERTY_CHANGE_PASSWORD;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+public class ChangePasswordActions extends LogInAction {
+    static Logger log = LoggerFactory.getLogger(ChangePasswordActions.class);
+    ChangePasswordObjects changePasswordObjects = new ChangePasswordObjects();
+    PropertyReader propertyChangePassword = new PropertyReader(PROPERTY_CHANGE_PASSWORD);
+    WebDriverWait wait = new WebDriverWait(driver, 10);
+    LogInPageObject logInPageObject = new LogInPageObject();
+    String newPassword = getNewPassword();
+
+    public void userLogsIn() {
+        logInPageObject.getLogInLink().click();
+        logInPageObject.getEmailInputField().sendKeys(propertyChangePassword.getPropertyValue("passUserName"));
+        logInPageObject.getPasswordInputField().sendKeys(propertyChangePassword.getPropertyValue("currentPassword"));
+        logInPageObject.getLogInButton().click();
+    }
+
+    public void navigateToChangePasswordPage() {
+        wait.until(ExpectedConditions.visibilityOf(changePasswordObjects.getUserEmail()));
```

Comment:
Why don't use wait method from GenericActions ?
Can be used as well in next steps